modules/p2/ding_entity_rating/ding_entity_rating.module

```diff
@@ -76,28 +76,31 @@ function ding_entity_rating_access($account) {
  * @param int $rating
  *   The given rating / weight.
  */
-function ding_entity_rating_add_to_list_callback($ajax, $entity, $account, $rating) {
-  // Access check for permission.
+function ding_entity_rating_add_to_list_callback($entity, $rating) {
   global $user;
-  if ($user->uid != $account->uid) {
-    return '';
-  }
   
-  if ($ajax) {
-    // Add to rated materials list.
-    ding_entity_rating_add_to_list($entity, $account, $rating);
-    
-    // Build response.
-    $output = theme('ding_entity_rating_thank_you', array());
-    
-    // Return response.
-    print $output;
-    drupal_exit();
+  $commands = array();
+  
+  // Grabbed from ding_reservation_reserve_ajax()
+  if (!user_is_logged_in()) {
+    // Trigger log-in (the reservation link will be triggered on success).
+    $commands[] = ajax_command_ding_user_authenticate('');
+  }
+  elseif (!ding_user_is_provider_user($user)) {
+    // Error not library user.
+    $commands[] = ajax_command_ding_popup('ding_reservation', t('Error'), '<p>' . t('Only library user can make reservations.') . '</p>');
   }
   else {
-    // Only ajax supported.
-    return '';
+    ding_entity_rating_add_to_list($entity, $user, $rating);
+    
+    $commands[] = ajax_command_ding_popup(
+      'ding_reservation',
+      t('Rated'), 
+      '<p>' . theme('ding_entity_rating_thank_you') . '</p>'
```

Comment:
calling the theme() function directly is strongly discouraged - see: https://api.drupal.org/api/drupal/includes!theme.inc/function/theme/7.x